katello/hooks/pre/30-upgrade.rb

```diff
@@ -37,6 +37,29 @@ def migrate_foreman
   Kafo::Helpers.execute('foreman-rake db:migrate')
 end
 
+def postgresql_10_upgrade
+  #Ensure there is at least 1x /var/lib/pgsql free disk space available on the main filesystem
+  POSTGRES_DIR = '/var/lib/pgsql'
+
+  begin
+    available_space = `df -k --total --exclude-type=tmpfs --output=avail`.split("\n")[-1].to_i
+    postgres_size = File.directory?(POSTGRES_DIR) ? `du -k -s #{POSTGRES_DIR}`.split[0].to_i : 0
+    if available_space < postgres_size
+      $stderr.puts "The postgres upgrade requires at least #{postgres_size} of storage."
+      kafo.class.exit 1
+    end
+  rescue StandardError
+    $stderr.puts 'Failed to verify available disk space'
+  end
+
+  Kafo::Helpers.execute('yum -y install rh-postgresql10-postgresql-server')
+  Kafo::Helpers.execute('foreman-maintain service stop')
+  Kafo::Helpers.execute('scl enable rh-postgresql10 -- postgresql-setup --upgrade')
+  Kafo::Helpers.execute('yum -y remove postgresql postgresql-server')
+  Kafo::Helpers.execute('rm -f /etc/systemd/system/postgresql.service')
```

Comment:
It surprises me that `yum remove postgresql-server` doesn't clean this up.

Reply:
Cause it gets overwritten by the module:

```
[root@centos7-katello-nightly-pg10 vagrant]# rpm -qlf /etc/systemd/system/postgresql.service 
file /etc/systemd/system/postgresql.service is not owned by any package
```

Reply:
Oh that's ugly. That should be using drop in files.